### PR TITLE
nimble/controller: Multiple fixes out of IOP testing

### PR DESCRIPTION
**Pull request by rymanluk**
This is set of patches done just before and on UPF58. mostly related to extended scanner and extended connection creation.

### CODE REVIEW

File: net/nimble/controller/src/ble_ll_scan.c [hunk 1]
```diff
@@ -1677,7 +1682,11 @@ ble_ll_scan_get_aux_data(struct ble_ll_scan_sm *scansm,
         (*aux_data)->chan = tmp_aux_data.chan;
         (*aux_data)->offset = tmp_aux_data.offset;
         (*aux_data)->mode = tmp_aux_data.mode;
-
+        if (has_addr) {
```
Comment (sjanc) [inline net/nimble/controller/src/ble_ll_scan.c]: this is allowed only on 1M and RFU on other PHYs

and we should probably handle TargetA in same way

Comment (rymanluk) [reply net/nimble/controller/src/ble_ll_scan.c]: Yes, there is Jira on this.

File: net/nimble/controller/src/ble_ll_scan.c [hunk 2]
```diff
@@ -201,7 +202,11 @@ ble_ll_aux_scan_cb(struct ble_ll_sched_item *sch)
     }
 
     STATS_INC(ble_ll_stats, aux_fired_for_read);
-    ble_phy_wfr_enable(BLE_PHY_WFR_ENABLE_RX, 0, BLE_LL_SCHED_ADV_MAX_USECS);
+
+    wfr_usec = scansm->cur_aux_data->offset_units ? 300 : 30;
+    /* Lets add one tick to make sure we will not miss aux */
+    wfr_usec += os_cputime_ticks_to_usecs(1);
```
Comment (andrzej-kaczmarek) [inline net/nimble/controller/src/ble_ll_scan.c]: this is hack and it should be marked appropriately

File: net/nimble/controller/src/ble_ll_scan.c [hunk 3]
```diff
@@ -2386,13 +2398,12 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
...
             ble_ll_scan_req_backoff(scansm, 1);
-            scansm->cur_aux_data = NULL;
         }
 
-        if (!BLE_LL_CHECK_AUX_FLAG(aux_data, BLE_LL_AUX_CHAIN_BIT)) {
-            /* If there is no chaining, we can remove data. Otherwise it is
-             * already scheduled for next aux*/
-            ble_ll_scan_aux_data_free(hdr->rxinfo.user_data);
+        if (BLE_LL_CHECK_AUX_FLAG(aux_data, BLE_LL_AUX_CHAIN_BIT)) {
+            // If there is chaining, we don't want to free aux data
```
Comment (andrzej-kaczmarek) [inline net/nimble/controller/src/ble_ll_scan.c]: c++ style comments are not allowed

File: net/nimble/controller/src/ble_ll_conn.c [hunk 4]
```diff
@@ -2918,16 +2918,23 @@ ble_ll_init_rx_pkt_in(uint8_t pdu_type, uint8_t *rxbuf, struct ble_mbuf_hdr *ble
...
+    }
+
+    if (CONN_F_AUX_CONN_REQ(connsm)) {
+        /* Wait for connection response */
+        if (pdu_type != BLE_ADV_PDU_TYPE_AUX_CONNECT_RSP) {
             return;
         }
+        ble_ll_scan_aux_data_free(ble_hdr->rxinfo.user_data);
     }
+
```
Comment (andrzej-kaczmarek) [inline net/nimble/controller/src/ble_ll_conn.c]: lots of extra empty lines in this commit...

File: net/nimble/controller/src/ble_ll_conn.c [hunk 5]
```diff
@@ -3113,7 +3117,7 @@ ble_ll_init_rx_isr_end(uint8_t *rxbuf, uint8_t crcok,
 #endif
 
         /* Ignore this packet - do not send to LL */
-        goto init_rx_isr_ignore_exit;
+        goto init_rx_isr_exit;
```
Comment (andrzej-kaczmarek) [inline net/nimble/controller/src/ble_ll_conn.c]: comment above should be also changed

File: net/nimble/controller/src/ble_ll_scan.c (hunk 2 above)
Comment (rymanluk) [reply net/nimble/controller/src/ble_ll_scan.c]: actually we don't need it anymore as ble_ll_sched_aux_scan() is fixed and takes rem_usec into account when scheduling. I will remove +1

File: net/nimble/controller/src/ble_ll_scan.c (hunk 3 above)
Comment (rymanluk) [reply net/nimble/controller/src/ble_ll_scan.c]: actually this whole if is not needed. Chain support will come in different PR.

**Pull request merged**